src/mantainer.rs

```diff
@@ -54,21 +54,24 @@ impl TableMantainer {
             debug!("TableMantainer::monitor_buckets woke up");
             let table_lock_read = ktable.read().await;
             let root = table_lock_read.root();
-            table_lock_read.idle_buckets().flat_map(|h| h.1).for_each(
-                |target| {
-                    outbound_sender
-                        .try_send((
-                            Message::FindNodes(
-                                root.as_header(),
-                                *target.id().as_binary(),
-                            ),
-                            vec![*target.value().address()],
-                        ))
-                        .unwrap_or_else(|op| {
-                            error!("Unable to send broadcast {:?}", op)
-                        });
-                },
-            );
+
+            let idles = table_lock_read
+                .idle_buckets()
+                .flat_map(|(_, target)| target)
+                .map(|target| {
```

Comment:
why a `map` after a `flat_map`? can't you just perform the operation on the second parameter of the `flat_map`?

Reply:
It appears to be necessary since rust does not allow using `await` in the iterators